test/test_cli.py

```diff
@@ -0,0 +1,75 @@
+import shutil
+import unittest
+import subprocess
+import os
+from app import application
+
+
+class TestApplication(unittest.TestCase):
+    def test_initialize_repo(self):
+        try:
+            shutil.rmtree("../bitcointranscripts", ignore_errors=True)
+            subprocess.run(["rm", "-rf", "test"])
+        except Exception as e:
+            self.fail(e)
+
+    def test_find_source_type(self):
+        video = application.find_source_type("B0HW_sJ503Y") == "video"
+
+        playlist = application.find_source_type("PLPQwGV1aLnTuN6kdNWlElfr2tzigB9Nnj") == "playlist"
+
+        audio = application.find_source_type(
+            "https://anchor.fm/s/12fe0620/podcast/play/32260353/https%3A%2F%2Fd3ctxlq1ktw2nl.cloudfront.net%2Fstaging"
+            "%2F2021-3-26%2Fdc6f12e7-a547-d872-6ef6-7acfe755a692.mp3") == "audio"
+        self.assertTrue(audio and playlist and video)
+
+    def test_download_audio_file(self):
+        audio = application.get_audio_file(
+            "https://anchor.fm/s/12fe0620/podcast/play/32260353/https%3A%2F%2Fd3ctxlq1ktw2nl.cloudfront.net%2Fstaging"
+            "%2F2021-3-26%2Fdc6f12e7-a547-d872-6ef6-7acfe755a692.mp3")
+        self.assertTrue(os.path.isfile(audio))
+        os.remove(audio)
+
+    def test_process_mp3(self):
+        result = application.process_mp3("testAssets/audio.mp3", "tiny")
+        with open("testAssets/transcript.txt", "r") as file:
+            data = str(file.read().strip("\n"))
+            result = result.strip(" ")
+            self.assertEqual(data, result)
+
+    def test_download_video_file(self):
+        url = "https://www.youtube.com/watch?v=B0HW_sJ503Y"
+        video = application.download_video(url)
+        self.assertTrue(os.path.isfile(video) and os.path.isfile(video[:-3] + "description"))
+        print()
+        os.remove(video)
+        os.remove(video[:-3] + "description")
+
+    def test_chapter_creation(self):
```

Comment:
This is much more inline with what I'd suggest. Testing features based on mocks.